tools/tplg_parser/tplg_parser.c

```diff
@@ -1193,44 +1201,51 @@ int load_widget(void *dev, int dev_type, struct comp_info *temp_comp_list,
 		if (load_pipeline(dev, comp_id, pipeline_id, widget,
 				  *sched_id) < 0) {
 			fprintf(stderr, "error: load pipeline\n");
+			free(widget);
 			return -EINVAL;
 		}
 		break;
 	case(SND_SOC_TPLG_DAPM_SRC):
 		if (load_src(dev, comp_id, pipeline_id, widget, tp) < 0) {
 			fprintf(stderr, "error: load src\n");
+			free(widget);
 			return -EINVAL;
 		}
 		break;
 	case(SND_SOC_TPLG_DAPM_ASRC):
 		if (load_asrc(dev, comp_id, pipeline_id, widget, tp) < 0) {
 			fprintf(stderr, "error: load src\n");
+			free(widget);
```

Comment:
I think we should instead create a error exit path using a goto so in case any additional mallocs are added we don't have to add this many "frees"

Reply:
Agree. Updated the commit to use ```goto``` to create an exit path where all the memory freeing as well as returning error code is done.